### ISSUE TITLE
[NODE] Multiple node requests (2 or more) are made for each price request triggered from remix.

### ISSUE THREAD

**Issue by ebadiere** [smartcontractkit/chainlink#4178]
**Description**
When I run a price request from remix against my oracle I run one request, confirm in metamask, and then I get two requests in my chainlink node.  

**Basic Information**
As mentioned above, one price request from remix triggers two oracle node requests. One errors out.  The other succeeds, so there is no data loss.  This was happening intermittently, but seems to be happening consistently today, Apr 5, 2021.  There is nothing odd in the logs.

- Network: Polygon mainnet
- Blockchain Client: I can reproduce this in remix with metamask, so I assume matic vigil?

https://docs.matic.network/docs/develop/metamask/config-matic/

Here is the solidity used to test:
```
pragma solidity ^0.6.0;

import "https://github.com/smartcontractkit/chainlink/blob/develop/evm-contracts/src/v0.6/ChainlinkClient.sol";
import "https://github.com/smartcontractkit/chainlink/blob/develop/evm-contracts/src/v0.6/vendor/Ownable.sol";

contract APIConsumer is ChainlinkClient, Ownable {
  
    uint256 public price;
    
    address private oracle;
    bytes32 private jobId;
    uint256 private fee;
    
    /**
     * Network: Matic Mainnet
 
     * LINK address: 0xb0897686c545045aFc77CF20eC7A532E3120E0F1
     * Fee: 0.00 LINK
     */
    constructor() public {
        setChainlinkToken(0xb0897686c545045aFc77CF20eC7A532E3120E0F1);
        fee = 10 ** 15; // 0.01 LINK
    }
    
    /**
     * Create a Chainlink request to retrieve API response, find the target price
     * data, then multiply by 100 (to remove decimal places from price).
     */
    function requestPrice(  
        address oracle, 
        string memory jobId, 
        string memory base, 
        string memory quote
        ) public returns (bytes32 requestId) 
    {
        Chainlink.Request memory request = buildChainlinkRequest(stringToBytes32(jobId), address(this), this.fulfill.selector);
        
        request.add("base", base);
        request.add("quote", quote);
        
        string[] memory path = new string[](1);
        path[0] = "result";
        request.addStringArray("copyPath", path);
        request.addInt("times", 1e10);        
        
        // Sends the request
        return sendChainlinkRequestTo(oracle, request, fee);
    }
    
    /**
     * Receive the response in the form of uint256
     */ 
    function fulfill(bytes32 _requestId, uint256 _price) public recordChainlinkFulfillment(_requestId)
    {
        price = _price;
    }
    
    function stringToBytes32(string memory source) private pure returns (bytes32 result) {
        bytes memory tempEmptyStringTest = bytes(source);
        if (tempEmptyStringTest.length == 0) {
            return 0x0;
        }

        assembly { // solhint-disable-line no-inline-assembly
            result := mload(add(source, 32))
        }
    }   

    function withdrawLink() public onlyOwner {
     LinkTokenInterface link = LinkTokenInterface(chainlinkTokenAddress());
     require(link.transfer(msg.sender, link.balanceOf(address(this))), "Unable to transfer");
  } 
}
```

Set up your metamask to use polygon mainnet.  We are also using the fiews failover proxy, with multiple public nodes.

```
docker run -d --name proxy -p 4000:4000 fiews/cl-eth-failover  wss://ws-mainnet.matic.network wss://rpc-mainnet.maticvigil.com/ws wss://matic-mainnet-full-ws.bwarelabs.com wss://matic-mainnet-archive-ws.bwarelabs.com
```

- Go Version: N/A, but we're running chainlink node version 0.10.3
- Operating System: 
```
root@ip-172-31-2-130:~# uname -a
Linux ip-172-31-2-130 5.4.0-1041-aws #43-Ubuntu SMP Fri Mar 19 22:06:16 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
``` 
Commit: 
```
"level":"info","ts":1617369105.9671834,"caller":"cmd/local_client.go:56","msg":"Starting Chainlink Node 0.10.3 at commit 5dbbc21f13fa00ee724c439273156834c45ab93e","id":"boot","Version":"0.10.3","SHA":"5dbbc21f13fa00ee724c439273156834c45ab93e","InstanceUUID":"f069afdb-64b9-47ab-bcca-50c438094d1c"}
```
- Hosting Provider: AWS, EC2 ubuntu, RDS for postgres, and lambda for the adapter.
- Startup Command: 
```
docker pull smartcontract/chainlink:0.10.3; docker kill primary; yes | docker system prune; cd ~/.chainlink && docker run --restart=always  -p 6689:6689 -d --name primary -v ~/.chainlink:/chainlink -it --env-file=.env smartcontract/chainlink:0.10.3 local n -p /chainlink/.password
```

**Environment Variables**
```
{"level":"debug","ts":1617369116.6589868,"caller":"cmd/local_client.go:242","msg":"Environment variables\nALLOW_ORIGINS: *\nBALANCE_MONITOR_ENABLED: true\nBLOCK_BACKFILL_DEPTH: 10\nBRIDGE_RESPONSE_URL: \nETH_CHAIN_ID: 137\nCLIENT_NODE_URL: https://localhost:6689\nDATABASE_TIMEOUT: \nDATABASE_MAXIMUM_TX_DURATION: 30m0s\nDEFAULT_HTTP_LIMIT: 32768\nDEFAULT_HTTP_TIMEOUT: 15s\nCHAINLINK_DEV: false\nENABLE_EXPERIMENTAL_ADAPTERS: false\nETH_BALANCE_MONITOR_BLOCK_DELAY: 1\nETH_DISABLED: false\nETH_FINALITY_DEPTH: 50\nETH_GAS_BUMP_THRESHOLD: 3\nETH_GAS_BUMP_TX_DEPTH: 10\nETH_GAS_BUMP_WEI: 5000000000\nETH_GAS_LIMIT_DEFAULT: 500000\nETH_GAS_PRICE_DEFAULT: 20000000000\nETH_HEAD_TRACKER_HISTORY_DEPTH: 100\nETH_HEAD_TRACKER_MAX_BUFFER_SIZE: 3\nETH_MAX_GAS_PRICE_WEI: 1500000000000\nETH_URL: ws://proxy:4000/\nETH_SECONDARY_URLS: []\nEXPLORER_URL: \nFEATURE_EXTERNAL_INITIATORS: false\nFEATURE_FLUX_MONITOR: true\nFEATURE_OFFCHAIN_REPORTING: false\nFLAGS_CONTRACT_ADDRESS: \nGAS_UPDATER_BLOCK_DELAY: 1\nGAS_UPDATER_BLOCK_HISTORY_SIZE: 24\nGAS_UPDATER_ENABLED: false\nGAS_UPDATER_TRANSACTION_PERCENTILE: 60\nINSECURE_FAST_SCRYPT: false\nTRIGGER_FALLBACK_DB_POLL_INTERVAL: 30s\nJOB_PIPELINE_PARALLELISM: 4\nJOB_PIPELINE_REAPER_INTERVAL: 1h0m0s\nJOB_PIPELINE_REAPER_THRESHOLD: 168h0m0s\nJSON_CONSOLE: true\nLINK_CONTRACT_ADDRESS: 0xb0897686c545045aFc77CF20eC7A532E3120E0F1\nLOG_LEVEL: debug\nLOG_SQL_MIGRATIONS: true\nLOG_SQL: false\nLOG_TO_DISK: true\nMAXIMUM_SERVICE_DURATION: 8760h0m0s\nMIN_INCOMING_CONFIRMATIONS: 0\nMIN_OUTGOING_CONFIRMATIONS: 0\nMINIMUM_SERVICE_DURATION: \nMINIMUM_CONTRACT_PAYMENT: 0.001000000000000000\nMINIMUM_REQUEST_EXPIRATION: 300\nOCR_BOOTSTRAP_CHECK_INTERVAL: 20s\nOCR_CONTRACT_TRANSMITTER_TRANSMIT_TIMEOUT: 10s\nOCR_DATABASE_TIMEOUT: 10s\nP2P_LISTEN_IP: 0.0.0.0\nP2P_LISTEN_PORT: 0\nOCR_INCOMING_MESSAGE_BUFFER_SIZE: 10\nOCR_OUTGOING_MESSAGE_BUFFER_SIZE: 10\nOCR_NEW_STREAM_TIMEOUT: 10s\nOCR_DHT_LOOKUP_INTERVAL: 10\nOCR_TRACE_LOGGING: false\nOPERATOR_CONTRACT_ADDRESS: \nCHAINLINK_PORT: 6688\nREAPER_EXPIRATION: 240h0m0s\nREPLAY_FROM_BLOCK: -1\nROOT: /chainlink\nSECURE_COOKIES: true\nSESSION_TIMEOUT: 15m0s\nCHAINLINK_TLS_HOST: \nCHAINLINK_TLS_PORT: 6689\nCHAINLINK_TLS_REDIRECT: false\n"}
```

**Steps to Reproduce**
1. Setup a node on polygon as described above.
2. Use the fiews failover proxy.  Not sure if this has anything to do with it. https://github.com/Fiews/ChainlinkEthFailover
3. Compile the solidity above in remix and deploy to polygon.
4. Fund the contract.
5. Run a price request.  

On the back end I am using the cryptocompare adapter, but we all saw it on various adapters and other solidity code triggered the behavior as well.  It feels as if the node is processing the same event message more than once.

**Additional Information**
[replace this line with any additional information you would like to provide, such as screenshots illustrating the issue]

**Comment by PatrickAlphaC** [comment]
Hmm... Thanks for making this! I haven't been able to reproduce. If you apply the same steps to kovan/rinkeby, do you get the same issue?

**Comment by ebadiere** [comment]
Per Rob || LinkRiver:

"The issue is not only about the total amount of requests the chainlink node sends during its normal interaction like explained before. These requests are "eth_getBlockByNumber" and "eth_getBalance". The issue appears while receiving  job requests, because the amount of full node requests per second is very high. This is a flush of requests during one/a few seconds which combines (eth_getBlockByNumber, eth_getBalance, eth_getTransactionReceipt, eth_sendRawTransaction and eth_getTransactionCount) .
The most third party providers or public accessible websockets provide a high amount of requests per day, but a too small number of requests per second (mostly 1-3), which is too low for quickly fulfilling a job request. You can relieve the node/ slow down the process if you set MIN_INCOMING_CONFIRMATIONS and MIN_OUTGOING_CONFIRMATIONS to 2 or higher in order to give the node more time to interact. We are going to publish  a medium article about the full technical implementation of the blockchain client connection to the chainlink node for high performance request interactions. I'm around to further discuss this matter if you want to exchange ideas folks!"

**Comment by PatrickAlphaC** [comment]
wow this is really helpful! Thank you! We will flag this.

**Comment by se3000** [comment]
I believe that this has been fixed. It should also be included in the latest 0.10.4 release. Please reopen if you see it in that release or later.

https://github.com/smartcontractkit/chainlink/pull/4204

**Comment by ebadiere** [comment]
I can't reopen this since you closed it and I am not a contributor.  We are running 0.10.4 and still seeing something like this.  We're getting ```TxHash 0x7fe210ec8b639a8697a8a9d7ca07f62c6847386da8445ff633b801a0fecef72c initiating run 8d349bc3-40b9-49fb-b7c3-ed5c1e773248 not on main chain; presumably has been uncled``` and ```transaction 0x754b97d1bd0e6362629ee3524e545f7316fd57aac31d4fd18d1ae0a0aaae33df reverted on-chain```.  Perhaps it's a separate issue but the behavior from a node operator's perspective looks the same.

**Comment by bamerrell** [comment]
We've been experiencing this too with our Kovan node.  Or experience doesn't seem to mesh with what Rob described, as we've reproduced it (somewhat randomly) using a Quiknode API key that supposedly doesn't rate limit per second, while also raising our incoming confirmations to 9. For this testing we've kept it simple and not used the failover proxy.

We are running 0.10.4 and using a v0.4 Oracle contract. If more details would be helpful let me know and I'm happy to provide.
